src/lib/Libutil/misc_utils.c

```diff
@@ -1520,3 +1532,221 @@ get_preemption_order(struct preempt_ordering *porder, int req, int used)
 
 	return po;
 }
+
+#ifdef WIN32
+/**
+ * @brief
+ *	Returns the current wall clock time.
+ *
+ * @return double - number of seconds.
+ */
+static double 
+get_walltime(void)
+{
+	LARGE_INTEGER time, freq;
+
+	if (QueryPerformanceFrequency(&freq) == 0)
+        	return (0);
+
+	if (QueryPerformanceCounter(&time) == 0)
+        	return (0);
+
+	return ((double)time.QuadPart / freq.QuadPart);
```

Comment:
Do we need to check freq.QuadPart to make sure it's not zero?

Reply:
Checked msdn/win32 info on QueryPerformanceFrequency():
https://msdn.microsoft.com/en-us/library/windows/desktop/ms644905(v=vs.85).aspx

It asserts:

BOOL WINAPI QueryPerformanceFrequency(
  _Out_ LARGE_INTEGER *lpFrequency
);

Parameters
lpFrequency [out]
A pointer to a variable that receives the current performance-counter frequency, in counts per second. **If the installed hardware doesn't support a high-resolution performance counter, this parameter can be zero (this will not occur on systems that run Windows XP or later).**

Return value
If the installed hardware supports a high-resolution performance counter, the return value is nonzero.

If the function fails, the return value is zero. To get extended error information, call GetLastError. On systems that run Windows XP or later, the function will always succeed and will thus never return zero.
--
Since we return early if QueryPerformanceFrequency()  has a return code of 0, then we should get to a divide by zero crash.